### PR TITLE
ChestMenu per player in UU category

### PR DESCRIPTION
**Pull request by Riley31415**
## Short Description
Made a chest menu per player in UU category

## Additions/Changes/Removals
Moved stuff around so that there is a chest menu per player in UU category

## Related Issues
uu category is weird with multiple people in it

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with base Slimefun and made sure nothing breaks/unexpected happens.
- [ ] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added which may not be obvious to maintainers.
- [ ] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,4 +5,5 @@
 *.iml
 .classpath
 .project
+dependency-reduced-pom.xml
```
Comment (J3fftw1) [inline .gitignore]: this should be .dependency-reduced-pom.xml

File: src/main/java/dev/j3fftw/litexpansion/uumatter/UuMatterCategory.java [hunk 2]
```diff
@@ -22,25 +22,12 @@
 
     public static final UuMatterCategory INSTANCE = new UuMatterCategory();
 
-    private final int[] recipeSlots = new int[] {12, 13, 14, 21, 22, 23, 30, 31, 32};
-    private ChestMenu menu;
+    private final int[] recipeSlots = new int[]{12, 13, 14, 21, 22, 23, 30, 31, 32};
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/uumatter/UuMatterCategory.java]: add the space back

File: src/main/java/dev/j3fftw/litexpansion/uumatter/UuMatterCategory.java [hunk 3]
```diff
@@ -22,25 +22,12 @@
...
 
     private UuMatterCategory() {
         super(new NamespacedKey(LiteXpansion.getInstance(), "uumatter_category"),
-            new CustomItem(Items.UU_MATTER, "&5UU-Matter Recipes")
+                new CustomItem(Items.UU_MATTER, "&5UU-Matter Recipes")
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/uumatter/UuMatterCategory.java]: formatting

File: src/main/java/dev/j3fftw/litexpansion/uumatter/UuMatterCategory.java [hunk 4]
```diff
@@ -58,7 +45,7 @@ private void displayItem(PlayerProfile profile, ItemStack result) {
             for (int i = 0; i < 9; i++) {
                 if (i == 1) {
                     menu.addItem(i, new CustomItem(ChestMenuUtils.getBackButton(p, "",
-                        ChatColor.GRAY + SlimefunPlugin.getLocalization().getMessage(p, "guide.back.guide")))
+                            ChatColor.GRAY + SlimefunPlugin.getLocalization().getMessage(p, "guide.back.guide")))
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/uumatter/UuMatterCategory.java]: formatting

File: src/main/java/dev/j3fftw/litexpansion/uumatter/UuMatterCategory.java [hunk 5]
```diff
@@ -87,7 +74,7 @@ private void displayItem(PlayerProfile profile, ItemStack result) {
 
     private void displayItem(ChestMenu menu, Player p, PlayerProfile profile, ItemStack output, ItemStack[] recipe) {
         final ChestMenu.MenuClickHandler clickHandler = (pl, s, clickedItem, a) -> onIngredientClick(profile,
-            clickedItem);
+                clickedItem);
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/uumatter/UuMatterCategory.java]: formatting

File: src/main/java/dev/j3fftw/litexpansion/uumatter/UuMatterCategory.java [hunk 6]
```diff
@@ -116,8 +103,17 @@ public boolean isVisible(@Nonnull Player player, @Nonnull PlayerProfile playerPr
...
+        // Header
+        for (int i = 0; i < 9; ++i) {
+            menu.addItem(i, ChestMenuUtils.getBackground(), ChestMenuUtils.getEmptyClickHandler());
+        }
+
+        menu.setEmptySlotsClickable(false);
+
         menu.addItem(1, new CustomItem(ChestMenuUtils.getBackButton(player, "",
-            ChatColor.GRAY + SlimefunPlugin.getLocalization().getMessage(player, "guide.back.guide")))
+                ChatColor.GRAY + SlimefunPlugin.getLocalization().getMessage(player, "guide.back.guide")))
```
Comment (J3fftw1) [inline src/main/java/dev/j3fftw/litexpansion/uumatter/UuMatterCategory.java]: formatting

**Review by J3fftw1 — CHANGES_REQUESTED** [review]
do you have a video of it working?

**Comment by NCBPFluffyBear** [comment]
> do you have a video of it working?

if you need help testing @Mooy1 just dm me ip

**Pull request merged**